contracts/libraries/VaultLifecycleSTETH.sol

```diff
@@ -508,13 +385,26 @@ library VaultLifecycleSTETH {
         address collateralToken,
         address recipient,
         uint256 amount
-    ) external returns (uint256 withdrawAmount) {
-        withdrawAmount = IWSTETH(collateralToken).getWstETHByStETH(amount);
+    ) external returns (uint256) {
+        IWSTETH collateral = IWSTETH(collateralToken);
+
+        uint256 withdrawAmount = collateral.getWstETHByStETH(amount);
 
         uint256 yieldTokenBalance =
             withdrawYieldToken(collateralToken, recipient, withdrawAmount);
 
-        // If there is not enough stETH in the vault, it withdraws as much as possible and
+        // If there is not enough wstETH in the vault, it withdraws as much as possible steth
+        if (withdrawAmount > yieldTokenBalance) {
+            yieldTokenBalance = yieldTokenBalance.add(
+                withdrawYieldToken(
+                    collateral.stETH(),
```

Comment:
I just noticed that this is a storage variable, there's no reason it should be because the address won't ever be changed. We can make it an immutable

Reply:
Only for the stETH vault, we can also make the `collateralToken` an immutable variable because it is never changed. Not for the Yearn vaults though

Reply:
steth as a storage variable? 

I changed collateralToken to immutable